### PR TITLE
docs(http-add-on): Introduce helm-docs for documentation

### PR DESCRIPTION
**Pull request by mkilchhofer**
Implements the same from PR #446 for the http-add-on.
Related to:
- #444

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] README is updated with new configuration values *(if applicable)*

### CODE REVIEW

File: http-add-on/values.yaml [hunk 1]
```diff
@@ -140,14 +137,17 @@ images:
...
+  # -- Image name for the interceptor image component
   interceptor: ghcr.io/kedacore/http-add-on-interceptor
-  # the scaler image to use. the default is the official build
+  # -- Image name for the scaler image component
   scaler: ghcr.io/kedacore/http-add-on-scaler
   # the kube-rbac-proxy image to use
   kubeRbacProxy:
+    # -- Image name for the Kube RBAC Proxy image component
     name: gcr.io/kubebuilder/kube-rbac-proxy
+    # -- Image tag for the Kube RBAC Proxy image componentq
```
Comment (JorTurFer) [inline http-add-on/values.yaml:152]:
```suggestion
    # -- Image tag for the Kube RBAC Proxy image component
```

Comment (mkilchhofer) [reply http-add-on/values.yaml]: fixed and rebased

**Comment by JorTurFer** [comment]
Thanks for this awesome contribution ❤️  This tool is fascinating
WDYT @kedacore/keda-helm-contributors ?

File: http-add-on/README.md [hunk 2]
```diff
@@ -80,66 +80,77 @@ The command removes all the Kubernetes components associated with the chart and
...
+| `operator.imagePullSecrets` | list | `[]` | The image pull secrets for the operator component |
+| `operator.nodeSelector` | object | `{}` | Node selector for pod scheduling ([docs](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/)) |
+| `operator.port` | int | `8443` | The port for the operator main server to run on |
+| `operator.pullPolicy` | string | `"Always"` | The image pull policy for the operator component |
+| `operator.resources.limits` | object | `{"cpu":0.5,"memory":"64Mi"}` | The CPU/memory resource limit for the operator component |
+| `operator.resources.requests` | object | `{"cpu":"250m","memory":"20Mi"}` | The CPU/memory resource request for the operator component |
+| `operator.tolerations` | list | `[]` | Tolerations for pod scheduling ([docs](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/)) |
+| `operator.watchNamespace` | string | `""` | The namespace to watch for new `HTTPScaledObject`s. Leave this blank (i.e. `""`) to tell the operator to watch all namespaces. |
+
+### scaler parameters
```
[the remark this answers is not in the document]
Comment (mkilchhofer) [reply http-add-on/README.md:111]: Yes sure we can customize that. We can even remove the separation completly but I thought that it is more clear when we group the related parameters into smaller sections.

WDYT?

[the remark this answers is not in the document]
Comment (tomkerkhove) [reply http-add-on/README.md:111]: I think the seperate categories are OK, but how do we customize them? :D

Can we go with `Scaler` or `Scaler parameters`?

[the remark this answers is not in the document]
Comment (mkilchhofer) [reply http-add-on/README.md:111]: Sure, I will update it.
The README.md is generated from README.md.gotmpl file. This one is almost the same except the parts where golang templating (curly brackets) are present.

[the remark this answers is not in the document]
Comment (mkilchhofer) [reply http-add-on/README.md]: Updated and rebased the PR :-)

**Review by tomkerkhove — APPROVED** [review]
Wonderful, thanks!

**Pull request merged**